frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -2005,6 +2061,33 @@ mod tests {
 		})
 	}
 
+	#[test]
+	fn challenge_solution_works() {
+		ExtBuilder::default().build_and_execute(|| {
+			crate::mock::SignedMaxSubmissions::set(50);
+			roll_to(14);
+			assert_eq!(MultiPhase::current_phase(), Phase::Off);
+
+			roll_to(15);
+			assert_eq!(multi_phase_events(), vec![Event::SignedPhaseStarted { round: 1 }]);
+			assert_eq!(MultiPhase::current_phase(), Phase::Signed);
+
+			let mut solution = raw_solution();
+
+			solution.score.minimal_stake += 1;
+
+			assert!(mock::Balances::usable_balance(&99) == 100);
+
+			assert_ok!(MultiPhase::submit(crate::mock::Origin::signed(99), Box::new(solution),));
+
+			assert_eq!(crate::mock::Balances::usable_balance(&9999), 100);
+
+			assert_ok!(MultiPhase::challenge_solution(crate::mock::Origin::signed(9999), 0));
+
+			assert!(crate::mock::Balances::free_balance(&9999) > 100);
```

Comment:
Please add a test about an unsuccessful challenge as well, and then you will see what I mean by the reward of the challengee being lost.